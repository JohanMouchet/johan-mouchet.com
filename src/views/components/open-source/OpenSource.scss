@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.c-open-source {
  &__lede {
    font-style: italic;
  }
}
