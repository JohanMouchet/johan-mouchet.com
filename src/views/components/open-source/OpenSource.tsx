import React from "react";
import {
  Elements,
  RichText,
  RichTextBlock,
  RichTextSpan,
} from "prismic-reactjs";
import { parse } from "utils/parse";
import "./OpenSource.scss";

type Props = {
  projects: {
    link: {
      url: string;
    };
    name: string;
    lede: RichTextBlock[];
    description: {
      type: Elements.paragraph | Elements.preformatted;
      text: string;
      spans: RichTextSpan[];
    }[];
  }[];
};

const OpenSource: React.FC<Props> = ({ projects }) =>
  !projects?.length ? null : (
    <div className="c-open-sources">
      <div className="grid">
        {projects.map((project) => {
          const descriptionPreformatted =
            project.description[0].type === Elements.preformatted;

          return (
            <div
              className="cell cell--12-@xs cell--6-@sm cell--4-@lg"
              key={project.name}
            >
              <h3 className="c-open-source__title">
                <a
                  href={project.link.url}
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  {project.name}
                </a>
              </h3>

              <div className="c-open-source__lede">
                <RichText render={project.lede} />
              </div>

              <div className="c-open-source__description">
                {descriptionPreformatted ? (
                  parse(project.description[0].text)
                ) : (
                  <RichText render={project.description} />
                )}
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );

export default OpenSource;
