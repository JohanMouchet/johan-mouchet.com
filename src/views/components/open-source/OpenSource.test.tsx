import React from "react";
import { render } from "@testing-library/react";
import { Elements } from "prismic-reactjs";
import OpenSource from ".";

it("Renders <OpenSource /> without crashing", () => {
  expect(
    render(
      <OpenSource
        projects={[
          {
            link: {
              url: "#",
            },
            name: "Name",
            lede: [
              {
                type: Elements.paragraph,
                text: "Lede",
                spans: [],
              },
            ],
            description: [
              {
                type: Elements.preformatted,
                text: "<ul><li>Description</li></ul>",
                spans: [],
              },
            ],
          },
        ]}
      />
    ).container.firstChild
  ).toMatchSnapshot();
});
