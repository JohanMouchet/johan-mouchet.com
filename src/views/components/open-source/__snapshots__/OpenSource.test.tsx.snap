// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <OpenSource /> without crashing 1`] = `
<div
  class="c-open-sources"
>
  <div
    class="grid"
  >
    <div
      class="cell cell--12-@xs cell--6-@sm cell--4-@lg"
    >
      <h3
        class="c-open-source__title"
      >
        <a
          href="#"
          rel="noopener noreferrer"
          target="_blank"
        >
          Name
        </a>
      </h3>
      <div
        class="c-open-source__lede"
      >
        <p>
          Lede
        </p>
      </div>
      <div
        class="c-open-source__description"
      >
        <ul>
          <li>
            Description
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
